test/rhsmlib/dbus/test_register.py

```diff
@@ -321,257 +312,162 @@
-
-    def test_can_get_orgs_over_domain_socket(self):
-        def assertions(*args):
-            # Be sure the returned json contains list of orgs
-            expected_owners = json.loads(OWNERS_CONTENT_JSON)
-            self.assertEqual(json.loads(args[0]), expected_owners)
-
-        self.mock_identity.is_valid.return_value = False
-        self.mock_identity.uuid = "INVALIDCONSUMERUUID"
-
-        dbus_method_args = ["admin", "admin", {}, ""]
-        self.dbus_request(assertions, self._build_interface().GetOrgs, dbus_method_args)
-
-    def test_can_register_over_domain_socket_with_activation_keys(self):
-        expected_consumer = json.loads(CONSUMER_CONTENT_JSON)
-
-        def assertions(*args):
-            # Be sure we are persisting the consumer cert
-            self.assertEqual(json.loads(args[0]), expected_consumer)
-
-        self.mock_identity.is_valid.return_value = False
-        self.mock_identity.uuid = "INVALIDCONSUMERUUID"
-
-        self.mock_register.register.return_value = expected_consumer
-
-        dbus_method_args = [
-            "admin",
+    def test_Stop(self):
+        self.obj.Start.__wrapped__(self.obj, self.LOCALE)
+        self.obj.Stop.__wrapped__(self.obj, self.LOCALE)
+
+    def test_Stop__not_running(self):
+        with self.assertRaises(rhsmlib.dbus.exceptions.Failed):
+            self.obj.Stop.__wrapped__(self.obj, self.LOCALE)
+
+
+class DomainSocketRegisterDBusObjectTest(DBusServerStubProvider):
+    dbus_class = DomainSocketRegisterDBusObject
+    dbus_class_kwargs = {}
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        register_patch = mock.patch(
+            "rhsmlib.dbus.objects.register.RegisterService.register",
+            name="register",
+        )
+        cls.patches["register"] = register_patch.start()
+        cls.addClassCleanup(register_patch.stop)
+
+        is_registered_patch = mock.patch(
+            "rhsmlib.dbus.base_object.BaseObject.is_registered",
+            name="is_registered",
+        )
+        cls.patches["is_registered"] = is_registered_patch.start()
+        cls.addClassCleanup(is_registered_patch.stop)
+
+        update_patch = mock.patch(
+            "rhsmlib.dbus.objects.register.EntCertActionInvoker.update",
+            name="update",
+        )
+        cls.patches["update"] = update_patch.start()
+        cls.addClassCleanup(update_patch.stop)
+
+        attach_auto_patch = mock.patch(
+            "rhsmlib.dbus.objects.register.AttachService.attach_auto",
+            name="attach_auto",
+        )
+        cls.patches["attach_auto"] = attach_auto_patch.start()
+        cls.addClassCleanup(attach_auto_patch.stop)
+
+        build_uep_patch = mock.patch(
+            "rhsmlib.dbus.base_object.BaseObject.build_uep",
+            name="build_uep",
+        )
+        cls.patches["build_uep"] = build_uep_patch.start()
+        cls.addClassCleanup(build_uep_patch.stop)
+
+        super().setUpClass()
+
+    def setUp(self) -> None:
```

Comment:
this needs to call `setUp()` of the parent class (`DBusServerStubProvider`), so in case we add one implementation in the future it will be used as expected